prime-router/src/main/kotlin/history/azure/HistoryDatabaseAccess.kt

```diff
@@ -178,15 +178,11 @@ abstract class HistoryDatabaseAccess(
             filter = filter.and(ACTION.CREATED_AT.lt(until))
         }
 
-        val failedFilter: Condition = when (showFailed) {
-            true -> {
-                ACTION.HTTP_STATUS.between(200, 600)
-            }
-            false -> {
-                ACTION.HTTP_STATUS.between(200, 299)
-            }
-        }
+        if (showFailed)
+            return filter
```

Comment:
Try not to have multiple returns.  Here you could bring up the return to the if and have an else.